content/blog/how-to-store-tokens-safely/index.md

```diff
@@ -0,0 +1,59 @@
+
+Cookies are a good tool for this scenario, but of course you should follow some guidelines to make it safe such as set the cookie as `secure`, `httpOnly` and `SameSite=strict`. The `secure` flag ensures that the cookie is sent only over `https` connections, the `httpOnly` makes the cookie inaccessible on client side so malicious scripts cannot see your sensitive data in `document.cookies` and `SameSite=strict` blocks cookies from being sent to an external domain.
+
+Since the cookie is not available on client side, we need to create some backend logic to fetch the token and store in memory. You can create a BFF(backend for frontends) using a web framework like Rails, Django, Phoenix, Laravel, etc. or you can use lambdas which are very easy and fast to move on. I recommend that you use Netlify Functions or Zeit. Let's see how it looks like.
+
+### Login Flow
+
+![Login Flow](./login-flow.png)
+
+In this flow what we are doing is creating a token and storing it in a httpOnly cookie to be protected by our server side preventing it to be accessed from malicious front-end/client scripts. After that, the app can stores the token in memory and use it to run your operations. 
+
+*Notice the function is acting as the cookie guardian where is stored the token.*
+
+You can see a more detailed explanation bellow:
+
+1. User access the login page.
+2. Send the credentials to login function.
+3. The login function validates the credentials and creates a token if the credentials are valid. If it is not, it returns an unauthorized error.
+4. If it is valid, the function saves the token in the cookie as httpOnly so, only servers can access it preventing it to be accessed from malicious front-end/client scripts.
+5. (6 and 7) The function returns the token and the app saves it in memory.
+
+
+Ok, so if my user refreshs the page it will loose the token and crash my app so what we should do? We need to validate the user session trough a validate function.
+
+### Validation Flow
+
+![Validate Flow](./validate-flow.png)
+
+In this flow we need to check if the user is already logged in. So the app send a request to the validate function which will check if there is any token stored in the cookie and if it is valid. If it is valid, the validate functions returns it to the app which will store it in the memory. Like we did before, I put it in a more descriptive way right bellow:
+
+1. User access the app.
+2. The app validates the user session.
+3. (4 and 5) The function check if there is a token and if it is valid. If it is not, it returns an unauthorized error.
+6. If it is valid, the function returns the token and the app saves it in memory. 
```

Comment:
```suggestion
6. If the token is valid, the function returns the token and the app saves it in memory. 
```

Reply:
Why does the function return the token? The client provided the token, and it is valid, so the client already has a valid token. Does the token change during validation? Is this performing the function of a nonce as well?